### PR TITLE
fi-as2-amna

### PR DESCRIPTION
**Pull request by amna-kamran**
## PR Title

fi-as2-amna

## Things you have improved in this PR

- Built the third screen
- Improved code structure
- Used animated container widget
- Fixed file names
- Broke down code into widgets and used part and part of
- Arranged images and colors into a different file and included them as utils
- Tried generalizing the code
- Switched loops with maps

## Checklist

- [x] Added PR title as per format i.e. `fi-a#-your-name`
- [x] You can name folders name noramlly e.g. widgets, constants, screens etc.
- [x] Files and Widgets names are per format i.e. `fi_a#_your_name_home.dart` and `FiA#YourNameHome` e.g `FiA2MHamzaHomeScreen`
- [ ] Get reviewer approval

## Screenshots (if applicable)
image
image
image
image

image
image

### CODE REVIEW

File: assets/amna/Login screen logo.png
Comment (mhmzdev) [inline assets/amna/Login screen logo.png:1]: Please use one of the files either the `.png` or the `.svg` one

File: lib/assesments/assessment_2/f1_as2_amna_dashboard.dart [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    List<Widget> iconWidgets = icons
+        .asMap()
+        .entries
+        .map((entry) => GestureDetector(
+              onTap: () => setState(() {
+                currentIndex = entry.key;
+              }),
+              child: SvgPicture.asset(
+                entry.value,
+                // ignore: deprecated_member_use
```
Comment (mhmzdev) [inline lib/assesments/assessment_2/f1_as2_amna_dashboard.dart]: remove this line its causing a warning to go away

File: pubspec.yaml
Comment (mhmzdev) [inline pubspec.yaml:1]: Please undo the changes in this file except for the assets thing

**Review by mhmzdev — CHANGES_REQUESTED** [review]
Very good job, just minor changes. Update it and let me know

File: lib/assesments/assessment_2/amna/fi_as2_amna_dashboard.dart [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+class FiAs2AmnaDashboard extends StatefulWidget {
+  const FiAs2AmnaDashboard({super.key});
+
+  @override
+  State<FiAs2AmnaDashboard> createState() => _F1As2AmnaDashboardState();
+}
+
+class _F1As2AmnaDashboardState extends State<FiAs2AmnaDashboard> {
+  List<Widget> views = [
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/fi_as2_amna_dashboard.dart]: better make this final as this isnt changing

File: lib/assesments/assessment_2/amna/fi_as2_amna_dashboard.dart [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    const FiAs2AmnaProfile(),
+    const FiAs2AmnaNotification(), // 0
+  ];
+  Widget smallWidth = const SizedBox(
+    width: 32.0,
+  );
+  int currentIndex = 0;
+  @override
+  Widget build(BuildContext context) {
+    List<String> icons = [
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/fi_as2_amna_dashboard.dart]: make this final as well if this isn't changing

File: lib/assesments/assessment_2/amna/fi_as2_amna_dashboard.dart [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import 'package:flutter_svg/svg.dart';
+
+class FiAs2AmnaDashboard extends StatefulWidget {
+  const FiAs2AmnaDashboard({super.key});
+
+  @override
+  State<FiAs2AmnaDashboard> createState() => _F1As2AmnaDashboardState();
+}
+
+class _F1As2AmnaDashboardState extends State<FiAs2AmnaDashboard> {
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/fi_as2_amna_dashboard.dart]: members defined at class level should start with a leading underscore
```_formKey```
so that they are encapsulated

File: lib/assesments/assessment_2/amna/screens/home/fi_as2_amna_home.dart [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import '../../widgets/fi_as2_amna_avatar.dart';
+
+part 'widget/_fi_as2_amna_card.dart';
+
+class FiAs2AmnaHome extends StatelessWidget {
+  const FiAs2AmnaHome({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    List<String> images = [
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/screens/home/fi_as2_amna_home.dart]: make this final please

File: lib/assesments/assessment_2/amna/screens/home/fi_as2_amna_home.dart [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      FiAs2AmnaAssets.girl_1,
+      FiAs2AmnaAssets.man_1,
+      FiAs2AmnaAssets.girl_2,
+      FiAs2AmnaAssets.man_1,
+      FiAs2AmnaAssets.girl_1,
+      FiAs2AmnaAssets.girl_2,
+    ];
+
+    List<Widget> cards = [
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/screens/home/fi_as2_amna_home.dart]: instead to repeating const just write
``` const List<Widget> cards = [_whatever, _whatever2];```

File: lib/assesments/assessment_2/amna/screens/home/widget/_fi_as2_amna_card.dart [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+          ],
+        ),
+        child: Padding(
+          padding: const EdgeInsets.all(20.0),
+          child: Column(
+            children: [
+              Row(
+                children: [
+                  const FiAs2Avatar(
+                    imagePath: FiAs2AmnaAssets.girl_1,
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/screens/home/widget/_fi_as2_amna_card.dart:30]: Avoid using snake case

File: lib/assesments/assessment_2/amna/screens/login/widget/_fi_as2_amna_button.dart [hunk 8]
```diff
@@ -0,0 +1,32 @@
+part of '../fi_as2_amna_login.dart';
+
+class _Button extends StatelessWidget {
+  final int i;
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/screens/login/widget/_fi_as2_amna_button.dart:4]: please use some other name, i doesn't define the purpose of this variable

File: lib/assesments/assessment_2/amna/screens/login/widget/_fi_as2_amna_button.dart [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  const _Button({required this.i, required this.value});
+
+  @override
+  Widget build(BuildContext context) {
+    return InkWell(
+      onTap: () {},
+      child: Container(
+        width: 330.0,
+        decoration: BoxDecoration(
+            color: i == 0 ? Colors.transparent : FiAs2AmnaColors.primaryColor,
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/screens/login/widget/_fi_as2_amna_button.dart:15]: instead of taking i as an int take it a bool, since there is no particular reason for having an int if ultimately it's being converted to a bool, 
or just take a widget as backgroundColor and add the color depending on that,
both of these approaches are cleaner than using an int

File: lib/assesments/assessment_2/amna/screens/login/widget/_fi_as2_amna_button.dart [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+            borderRadius: BorderRadius.circular(40),
+            border: Border.all(color: Colors.grey)),
+        child: Padding(
+          padding: const EdgeInsets.all(25.0),
+          child: Text(
+            value,
+            textAlign: TextAlign.center,
+            style: TextStyle(
+                color: i == 0 ? Colors.black : Colors.white,
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/screens/login/widget/_fi_as2_amna_button.dart:24]: same here, you can have a bool hasBackgroundColor, border etc. and use that,

File: lib/assesments/assessment_2/amna/screens/login/fi_as2_amna_login.dart [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+import 'package:flutter_svg/svg.dart';
+import '../../theme/fi_as2_amna_constants.dart';
+part 'widget/_fi_as2_amna_button.dart';
+
+class F1As2AmnaLoginScreen extends StatelessWidget {
+  const F1As2AmnaLoginScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    Widget smallSpace = const SizedBox(
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/screens/login/fi_as2_amna_login.dart]: you can define and initialize this outside build, since build can be called multiple times

File: lib/assesments/assessment_2/amna/screens/profile/widget/fi_as2_amna_profile_card.dart [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+                          fontSize: 16,
+                          fontWeight: FontWeight.w600,
+                        ),
+                      ),
+                    ),
+            ))
+        .toList();
+    List<Widget> widgetsWithSeparator =
+        columns.expand((widget) => [widget, separator]).toList();
+    widgetsWithSeparator.add(SvgPicture.asset(FiAs2AmnaAssets.more)); //
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/screens/profile/widget/fi_as2_amna_profile_card.dart]: Any particular reason for the // here ? If not please remove it

File: lib/assesments/assessment_2/amna/screens/profile/widget/fi_as2_amna_stats.dart [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    return Padding(
+      padding: const EdgeInsets.only(right: 16, left: 16),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: buildStatBar(smallWidth),
+      ),
+    );
+  }
+
+  List<Widget> buildStatBar(Widget separator) {
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/amna/screens/profile/widget/fi_as2_amna_stats.dart:21]: instead of segregating widgets as functions make classes, functions cloud stack when build gets called again and again, making the app slow, classes do not have this problem,

**Review by ahmadexe — CHANGES_REQUESTED** [review]
Stuff like smallSpace, mediumSpace etc which you are repeating you can make them in a seperate file and follow DRY (Dont repeat yourself),

File: lib/assesments/assessment_2/amna/screens/login/widget/_fi_as2_amna_button.dart (hunk 9 above)
Comment (amna-kamran) [reply lib/assesments/assessment_2/amna/screens/login/widget/_fi_as2_amna_button.dart:15]: I need i as an int for this part. 
`  const _Button(
              i: 0,
              value: "Login",
            ),
            const _Button(
              i: 1,
              value: "Signup",
            )`
